src/backend/gporca/libgpopt/src/translate/CTranslatorExprToDXL.cpp

```diff
@@ -3910,6 +3910,7 @@ CTranslatorExprToDXL::PdxlnCorrelatedNLJoin(
 	// ORCA generates a plan with CorrelatedInnerNLJoin with a Const true join filter and a LIMIT over the
 	// scan of bar. The same foo.a = foo.b condition is also added as a filter on the table scan of foo.
 	if (CUtils::FScalarConstTrue(pexprScalar) &&
+		Edxlsubplantype(pexpr) == EdxlSubPlanTypeScalar &&
 		(COperator::EopPhysicalCorrelatedInnerNLJoin == op_id ||
```

Comment:
The comment above contains the `select * from foo where foo.a in (select foo.b from bar);` query as an example. The subplan from the query is not scalar. Do we need to change the comment?

Do we really need to call `Edxlsubplantype`? The function returns EdxlSubPlanTypeScalar when op_id is EopPhysicalCorrelatedInnerNLJoin. What about removing `COperator::EopPhysicalCorrelatedInLeftSemiNLJoin == op_id` from the condition?